data/src/main/java/com/microsoft/azure/kusto/data/KustoResultMapper.java

```diff
@@ -0,0 +1,106 @@
+package com.microsoft.azure.kusto.data;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.BiConsumer;
+import java.util.function.Supplier;
+
+/**
+ * A class for mapping Kusto results to a list of pojos
+ * 
+ * @param <R>
+ *            pojo type returned by the mapping
+ */
+public class KustoResultMapper<R> {
```

Comment:
I like the example code you included in the PR's description. Would you please include that as a few unit tests?

Reply:
done. See `KustoResultMapperTest`.